### PR TITLE
made interface for user service

### PR DESCRIPTION
**Pull request by darrentran**
- Created an interface to abstract the user service. 
- App performs auto login

### CODE REVIEW

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/fragments/LoginFragment.java [hunk 1]
```diff
@@ -24,7 +25,7 @@
  * the login button atttemps to log the user in. The fragment also contains a clickable
  * TextView that changes from a login fragment to a register fragment
  */
-public class LoginFragment extends Fragment implements UserService.LoginListener {
+public class LoginFragment extends Fragment implements IUserServiceProvider.LoginListener {
 
     EditText email, password;
     TextView signupText;
```
Comment (bojotamara) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/fragments/LoginFragment.java:31]: The member should be of type IUserServiceProvider, not UserService

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/fragments/RegisterFragment.java [hunk 2]
```diff
@@ -27,7 +28,7 @@
  * The fragment also contains a clickable TextView that changes from a register
  * fragment to a login fragment
  */
-public class RegisterFragment extends Fragment implements UserService.RegistrationListener {
+public class RegisterFragment extends Fragment implements IUserServiceProvider.RegistrationListener {
 
     private EditText fullname, username, email, password, confirmPassword;
     private TextView loginText;
```
Comment (bojotamara) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/fragments/RegisterFragment.java:34]: Same thing in this file

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/IUserServiceProvider.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+     * with the flow of the login process
+     */
+    interface LoginListener {
+        void onLoginSuccess();
+        void onLoginFailure();
+    }
+
+    /**
+     * This function is called to check if there is an existing instance of the logged in user.
+     * @return none
```
Comment (WJXHenry) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/IUserServiceProvider.java]: Remove the `@return` if it is a void function

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/IUserServiceProvider.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    /**
+     * This function first checks if the username is taken. If the username is not taken,
+     * the account will be created, otherwise it will notify the user that their username is already taken
+     * @param username Username to register with.
+     * @param email Email to register with.
+     * @param password Password of Account to register with.
+     * @param fullname Full name of user registering.
+     * @param listener Registration listener passed from fragment
+     * @return none
```
Comment (WJXHenry) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/IUserServiceProvider.java]: Remove `@return`

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/IUserServiceProvider.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+     */
+    void createUser(final String username, final String email, String password, final String fullname, final RegistrationListener listener);
+
+    /**
+     * This function is called to put the registered user into the database.
+     * @param username Username to register with.
+     * @param email Email to register with.
+     * @param fullname Full name of user registering.
+     * @param listener Registration listener passed from fragment
+     * @return none
```
Comment (WJXHenry) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/IUserServiceProvider.java:67]: Remove `@return`

File: Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/IUserServiceProvider.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+     */
+    void putUserIntoDB(final String email, final String fullname, final String username, final RegistrationListener listener);
+
+    /**
+     * This is a wrapper function that is called to login a user with their email and password.
+     * @param txt_email Email to login with.
+     * @param txt_password Password to login with.
+     * @param listener Login listener passed from fragment
+     * @return none
```
Comment (WJXHenry) [inline Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/IUserServiceProvider.java]: Remove `@return`

**Review by WJXHenry — DISMISSED** [review]
Minor issue with the Javadoc for `IUserServiceProvider`.

**Review by bojotamara — DISMISSED** [review]

**Review by bojotamara — APPROVED** [review]

**Pull request merged**